api/s2n.h

```diff
@@ -502,6 +502,35 @@ extern int s2n_get_cert_der(const struct s2n_cert *cert, const uint8_t **out_cer
 S2N_API
 extern int s2n_connection_get_peer_cert_chain(const struct s2n_connection *conn, struct s2n_cert_chain_and_key *cert_chain);
 
+/**
+ * Returns the DER encoding of an ASN.1 X.509 certificate extension value, it's length and a boolean critical.
+ * 
+ * Note that the memory for `oid_value_out` MUST be allocated by the caller. 
+ * 
+ * @param cert A pointer to the s2n_cert object being read.
+ * @param oid_field_in A pointer to a buffer that contains the OID of the X.509 certificate extension to be read.
+ * @param oid_field_in_len represents the length of the input buffer `oid_field_in`.
+ * @param oid_value_out A pointer to the output buffer which will hold the DER encoding of an ASN.1 X.509 certificate extension value returned. 
+ * @param oid_value_out_len  This return value represents the length of the output buffer `oid_value_out`.
+ * @param critical This return value contains the boolean value for `critical`.
+ */
+S2N_API 
+extern int s2n_get_x509_extension_oid_value(struct s2n_cert *cert, const uint8_t *oid_field_in, const uint32_t oid_field_in_len,
+                                      uint8_t **oid_value_out, uint32_t *oid_value_out_len, bool *critical);
+
+/**
+ * Returns the UTF8 String representation of the DER encoded ASN.1 X.509 certificate extension data. 
+ * 
+ * Note that the memory for `oid_value_out` MUST be allocated by the caller. 
```

Comment:
How can a caller know how much memory to allocate?

(We can tell them to allocate at least `extension_len - 2` which will always be enough.)